torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -259,12 +263,31 @@ void ProcessGroupAgent::sync() {
 }
 
 void ProcessGroupAgent::start() {
+  start_.store(true);
   listenerThread_ = std::thread(&ProcessGroupAgent::listenLoop, this);
 }
 
+void ProcessGroupAgent::localShutdown() {
+  if (!start_.load()) {
+    return;
+  }
+  LOG(INFO) << "Stopping ProcessGroupAgent.";
+  // CV to make sure we don't abort it in an invalid state.
+  {
+    std::unique_lock<std::mutex> lock(recvWorkMutex_);
+    recvWorkCV_.wait(lock, [this]() { return recvWork_; });
+    recvWork_->abort();
+    start_.store(false);
+  }
+  threadPool_.waitWorkComplete();
+  listenerThread_.join();
+  PythonRpcHandler::getInstance().cleanup();
```

Comment:
We need to call `_destroy_rref_context()` too. If you see `def join_rpc()`.

Reply:
I removed this since it was causing RRef leakage issues.